File: 2662_count_subarrays_where_max_element_appears_at_least_k_times.py
class Solution(object):
    def countSubarrays(self, a, k):
        m = max(a); l = c = res = 0
        for r in range(len(a)):
            c += a[r] == m
            while c >= k:
                res += len(a) - r
                c -= a[l] == m; l += 1
        return res

File: 2662_count_subarrays_where_max_element_appears_at_least_k_times.java
class Solution {
    public long countSubarrays(int[] z, int k) {
        int m = 0, l = 0, c = 0;
        for (int n : z) m = Math.max(m, n);
        long q = 0;
        for (int r = 0; r < z.length; r++) {
            if (z[r] == m) c++;
            while (c >= k) {
                q += z.length - r;
                if (z[l++] == m) c--;
            }
        }
        return q;
    }
}
